azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -57,8 +57,6 @@
 
 
 class HostPluginProtocol(object): # pylint: disable=R0902
-    _is_default_channel = False
```

Comment:
The repercussion of removing this property is that incase of a proxy, we would still always try direct first for every request and hence add additional delays per request. This can add up to a lot of unnecessary waiting and might end up adding valuable provision time for the VM (which will lead to TDPR issues for us). I think a better solution would be to keep our current logic where we set the HostGA to default and try HostGA first and if it fails, retry with direct rather than not do it at all. 

So essentially - 

try direct -> Pass (very good) -> Stop
                -> Fail -> Try HostGA -> Set HostGA as default

Next request, 
If HostGA default, try HostGA first -> Fail -> Try Direct again (we dont have this currently)

We can sync more about this offline if you want

Reply:
>always try direct first for every request and hence add additional delays per request. This can add up to a lot of unnecessary waiting and might end up adding valuable provision time for the VM
I thought about this and that's why I clarified in the description of this PR that your change #1967 significantly reduced the overall time it takes for a request to finally fail (after all retries).

I think it's better to remove a "default state" from HGAP because:
1) keeping state is dangerous in our not-so-safe multithreaded code
2) Windows does no such thing and we should converge the two behaviours. I believe they have a similar retry logic to ours, but I have to double check that.
3) Purely from a supportability perspective, it's better to be deterministic about our scenarios instead of having it rely on a case by case basis. Downloading manifests? Always direct first, then HGAP, not "depends on what the default is, which depends on what happened potentially a long time ago when one direct request failed".

These are my arguments for removing this altogether but I'm happy to discuss more.

Reply:
Plus, the concern to adding more time to TDPR is unchanged. There is no scenario where the HGAP would already be set as default during provisioning since there were no previous requests to fail, correct?

Reply:
Posting here for continuity:
>2. Windows does no such thing and we should converge the two behaviours. I believe they have a similar retry logic to ours, but I have to double check that.

This is true, but the important difference is that Windows always uses HGAP first, then direct. It's worth exploring that change in this PR too, I will do it shortly.

>Plus, the concern to adding more time to TDPR is unchanged. There is no scenario where the HGAP would already be set as default during provisioning since there were no previous requests to fail, correct?

I oversaw that the provisioning steps contains a lot of requests, so before my change one of the requests would fail, then we would be using HGAP throughout (meaning less retries/time to fail). If we change our behaviour to use HGAP first, we achieve the same thing.